apps/aecore/lib/aecore/structures/data_tx.ex

```diff
@@ -3,17 +3,28 @@ defmodule  Aecore.Structures.DataTx do
   Aecore structure of a transaction data.
   """
 
+  alias Aecore.Structures.DataTx
   alias Aecore.Keys.Worker, as: Keys
+  alias Aecore.Chain.ChainState
+  alias Aecore.Structures.SpendTx
   alias Aeutil.Serialization
   alias Aeutil.Parser
 
+  require Logger
+
+  @typedoc "Name of the specified transaction module"
+  @type txType :: SpendTx | atom()
+
+  @typedoc "Structure of a transaction that may be added to be blockchain"
+  @type payload :: SpendTx.t() | map()
```

Comment:
why do we need the `| map()`, should be removed or specified as type

Reply:
Same goes for the payload, we don't want to specificaly say that the payload can only be SpendTx

Reply:
what is this something else? should be specified in the type, if it for now is only `SpendTx`, just allow `SpendTx`

Reply:
Okay